python/lsst/resources/http.py

```diff
@@ -682,19 +742,35 @@ def _send_webdav_request(
         if url is None:
             url = self.geturl()
 
-        for _ in range(max_redirects := 5):
-            resp = self.session.request(
-                method, url, data=body, headers=headers, stream=True, timeout=TIMEOUT, allow_redirects=False
-            )
-            if resp.is_redirect:
-                url = resp.headers["Location"]
-            else:
-                return resp
+        with time_this(
+            log,
+            msg="%s %s",
+            args=(
+                method,
+                url,
+            ),
+            mem_usage=True,
+            mem_unit=u.mebibyte,
```

Comment:
Are you worried about memory usage?

Reply:
The intention is that we collect as much resource consumption information as we can when we are timing the execution. I thought it would not cost too much since `time_this` does not collect the memory usage when the logger is not enable for  `DEBUG` level, which is the normal production configuration.